delfin/task_manager/scheduler/schedule_manager.py

```diff
@@ -55,6 +62,40 @@ def start(self):
             self.scheduler.start()
             self.scheduler_started = True
 
+    def on_node_join(self, event):
+        # A new node joined the group, all the job would be re-distributed.
+        # If the job is already on the node, it would be ignore and would
+        # not be scheduled again
+        LOG.info('Member %s joined the group %s' % (event.member_id,
+                                                    event.group_id))
+        # Get all the jobs
+        tasks = db.task_get_all(self.ctx)
+        distributor = TaskDistributor(self.ctx)
+        partitioner = ConsistentHashing()
+        partitioner.start()
+        for task in tasks:
+            # Get the specific executor
+            origin_executor = task['executor']
+            # If the target executor is different from current executor,
+            # remove the job from old executor and add it to new executor
+            new_executor = partitioner.get_task_executor(task['id'])
+            if new_executor != origin_executor:
+                LOG.info('Re-distribute job %s from %s to %s' %
+                         (task['id'], origin_executor, new_executor))
+                self.task_rpcapi.remove_job(self.ctx, task['id'],
+                                            task['executor'])
+            distributor.distribute_new_job(task['id'])
```

Comment:
The scenario I considered is that our db recorded it in executor A, but actually it didn't run on executor A.
So, IMO, I think this should be outside the if block, because even the executor is same as before, we need to confirm that the executor already run the task, so I think we can send it again. If the executor already has the task, it would be ignored and have no side effect.